### PR TITLE
[BUG #2228] Implemented wallet filtering screen

### PR DESCRIPTION
**Pull request by jeluard**
addresses #2228 

### Summary:

Added filtering capacity to wallet transactions list

### Steps to test:
- Open Status
- Navigate wallet/transactions
- Verify filtering works as expected

status: ready

### CODE REVIEW

**Comment by yenda** [comment]
Isn't it a feature rather than a bug?

**Comment by jeluard** [comment]
@yenda Are you talking about the commit title? What does that change?

**Comment by yenda** [comment]
@jeluard doesn't change much, just fits better, I think it helps with development updates to distinguish features from bugs. I was talking about the PR title.

File: src/status_im/constants.cljs
Comment (flexsurfer) [inline src/status_im/constants.cljs]: i18n for labels?

File: src/status_im/ui/components/checkbox/view.cljs
Comment (flexsurfer) [inline src/status_im/ui/components/checkbox/view.cljs:13]: is this PR  suitable to change icon on svg here?

File: src/status_im/ui/screens/db.cljs
Comment (flexsurfer) [inline src/status_im/ui/screens/db.cljs:37]: why to not add in wallet map? `:wallet   {:transactions        constants/default-wallet-transactions}`

File: src/status_im/constants.cljs
Comment (flexsurfer) [inline src/status_im/constants.cljs:29]: looks like this is not a constant, because `:checked?` can be changed

File: src/status_im/ui/screens/wallet/transactions/views.cljs
Comment (flexsurfer) [inline src/status_im/ui/screens/wallet/transactions/views.cljs]: ?

**Review by flexsurfer — CHANGES_REQUESTED** [review]

File: src/status_im/constants.cljs
Comment (jeluard) [reply src/status_im/constants.cljs:29]: This is the default used when loading the app. Every restart loads this default. 
Can you think about another way to do this?

File: src/status_im/constants.cljs
Comment (jeluard) [reply src/status_im/constants.cljs]: Makes sense

File: src/status_im/ui/screens/db.cljs
Comment (jeluard) [reply src/status_im/ui/screens/db.cljs:37]: That's an option. Is the latter better, or our convention?

File: src/status_im/ui/screens/wallet/transactions/views.cljs
Comment (jeluard) [reply src/status_im/ui/screens/wallet/transactions/views.cljs]: Will be used when ERC20 support is added so I kept it.

File: src/status_im/ui/components/checkbox/view.cljs
Comment (jeluard) [reply src/status_im/ui/components/checkbox/view.cljs:13]: Good point

File: src/status_im/ui/screens/wallet/transactions/views.cljs
Comment (flexsurfer) [reply src/status_im/ui/screens/wallet/transactions/views.cljs]: could you add a comment in the code?

Comment (jeluard) [reply src/status_im/ui/screens/wallet/transactions/views.cljs]: ok

File: src/status_im/ui/screens/db.cljs
Comment (flexsurfer) [reply src/status_im/ui/screens/db.cljs:37]: i think @yenda has something to say here :)

**Review by flexsurfer — APPROVED** [review]

File: src/status_im/ui/screens/db.cljs
Comment (yenda) [reply src/status_im/ui/screens/db.cljs:37]: I think it is better to have the wallet related stuff in the wallet map yes

**Review by oskarth — APPROVED** [review]
Approve with minor comments that have already been mentioned (thumbs up)

File: src/status_im/ui/components/checkbox/view.cljs
Comment (jeluard) [reply src/status_im/ui/components/checkbox/view.cljs:13]: Actually I will skip that for now as checkboxes are first class citizen in latest RN so I can remove it completely.

File: src/status_im/ui/screens/db.cljs
Comment (jeluard) [reply src/status_im/ui/screens/db.cljs:37]: Not straightforward now as we cleanup the whole `wallet` key while initializing app-db.
Not quite sure what would be the proper way to address that so will skip for now.

Comment (oskarth) [reply src/status_im/ui/screens/db.cljs:37]: Is the reason we are cleaning up the whole wallet key not because you might switch account? Wouldn't that imply that we want transactions cleared as well then? Or am I misunderstanding?

Comment (yenda) [reply src/status_im/ui/screens/db.cljs:37]: Yes why is it a problem ? If there is default filters in default app-db there is nothing to do

Comment (jeluard) [reply src/status_im/ui/screens/db.cljs:37]: @oskarth What I want to store in `:transactions` is mostly static with some nested transient boolean. I expect the whole structure to be always there.
@yenda  I don't know why but my key disappear (not `transactions` as it is already used). Not investigated much as I want to merge this quick.

Comment (oskarth) [reply src/status_im/ui/screens/db.cljs:37]: If we don't merge it this week we got to create a whole swarm, oh jeez! :P I'm fine with this (PR, not creating a whole swarm for this).

Comment (yenda) [reply src/status_im/ui/screens/db.cljs:37]: @jeluard I don't understand what you tried, if you put the transaction key in the empty :wallet map in the default app-db (the line above the line this thread refers to), it will be there when you switch account.

Comment (jeluard) [reply src/status_im/ui/screens/db.cljs:37]: Not related to switching account, it wasn't there at all for some reason

Comment (yenda) [reply src/status_im/ui/screens/db.cljs:37]: add a todo if you don't want to spend the time on it

**Comment by asemiankevich** [comment]
Poor performance observed both for iOS and Android devices. Please check TF https://app.testfairy.com/projects/4803622-status/builds/7145145/sessions/2/?accessToken=MvgbcBW1DzCdos6PbE5dT1HN45o

STR:
1. go to transaction history screen - click filter
2. tick/untick checkboxes

Actually it takes ~2 seconds for the checkbox to reflect the changes (as well as for the buttons Select All and Done). @jeluard is looking into it now

**Comment by jeluard** [comment]
@asemiankevich I did some perf tests and can't see anything obvious due to this particular feature. This is most probably part of the global perf issues we have and that will be tackled in a separate `idea`.

I increased the tapping area of checkboxes as they were rather tiny.

**Comment by jeluard** [comment]
@asemiankevich I'd like to go ahead and merge this. What do you think?

**Comment by asemiankevich** [comment]
@jeluard did you removed the `postponed` filter from the options as agreed?

**Comment by jeluard** [comment]
@asemiankevich Now yes

**Comment by asemiankevich** [comment]
@jeluard please create last branch for this PR then so I will go through it before merging

**Pull request merged**
